rfcs/023-images-endpoint/README.md

```diff
@@ -0,0 +1,48 @@
+# RFC 023: Images endpoint
+
+**Status:** :construction: Draft :building_construction:
+
+**Last updated:** 27/04/2020
+
+## Context and requirements
+
+- Images are now entities in the catalogue API
+- We want to be able to search for images, and to fetch specified images
+- The images endpoint should behave similarly to the works endpoint
+
+## Proposal
+
+The initial images endpoint will be in **beta**, and therefore will be liable to have breaking changes made without a version bump.
+
+### Image display model
+
+```
+{
+  "id": String,
+  "location": DigitalLocation,
+  "parentWork": String
+}
+```
+
+This is intentionally minimal at this point as we decide what fields an image should have.
+
+### Single image
+
+**`GET`** `/catalogue/v2/images/:image_id`
+
+*Query params*
+
+- `include` - optional list of fields, as with works endpoint
+
+### Multiple images
+
+**`GET`** `/catalogue/v2/images`
+
+*Query params*
+
+- `query` - text search query
+- `similarTo` - an image ID; returned images are sorted by visual similarity to this.
+- `include` - as with the single image endpoint
+- `license` - license to filter for
```

Comment:
Does the DigitalLocation field include the license? If not, it feels a little odd that we'd filter by this but you'd not see it.

Reply:
It does, yes - I did wonder about extracting it but thought it was a better idea to keep it exactly the same as the works display model. Open to discussion/change though.

Reply:
In the works API we often use the path of the field to perform filtering, i.e. `location.license=...`. Don't know how consistent we are about that though.

Reply:
Looks like we're very inconsistent 🙃 - the license filter for works is filtering on (any) item's location's license https://github.com/wellcomecollection/catalogue/blob/master/api/api/src/main/scala/uk/ac/wellcome/platform/api/services/ElasticsearchRequestBuilder.scala#L143

Reply:
The field we try to use as the parameter is in the display JSON not Elastic path.

[So license as a queryparam](https://github.com/wellcomecollection/catalogue/blob/master/api/api/src/main/scala/uk/ac/wellcome/platform/api/rest/WorksParams.scala#L134) should probably have been `items.locations.license=bloop`. 🙃 ❗

Reply:
Indeed - if we're including the license here (and I think we should and support filtering by license), then we should follow the pattern from works:

- license goes on the location
- the query parameter is `locations.license`

Reply:
The discussion above is ambiguous - to clarify, license does go on location, but for the works endpoint the query parameter is just `license`, although as @jamesgorrie says it should've been `items.locations.license`.

What do you think we should go for here?

Reply:
The path version, whatever the path ends up being for `/images` - we should really fix this on `/works`